src/lib.rs

```diff
@@ -411,6 +408,10 @@ pub trait PathInfo {
         Path::extension(self.as_path())
     }
 
+    fn join<P: AsRef<Path>>(&self, path: P) -> PathBuf {
```

Comment:
@Screwtapello I added join here. Let me know if you have major objections. As much as I don't like adjoinment, not being API compatible is not something that can stand

Reply:
Is being API compatible really a big deal? I feel like `path_abs` has strong opinions about how path manipulation *should* work, making it easy to do things the "right" way and harder to do things the "wrong" way. If you're not a fan of adjoinment, don't put the method in, and see if you can get along without it. If somebody *really* needs adjoinment, `p.as_path().join()` isn't hard to say. Of course, it's up to you.

Apart from the whole adjoinment issue, this method should probably live on `PathOps` so that different types can produce different return types. `<PathBuf as PathOps>::join` should return a `PathBuf` because the result might be absolute or not, but `<PathAbs as PathOps>::join` should return a `PathAbs` because the result should always be absolute (even if not canonical).

Reply:
Not on windows. `PathAbs::new("C:/foo").join("/")` would not count as an absolute path in this library.

I need to consider it more. It is a good point that it doesn't really fit with the rest of the library. At the same time, I want to reduce _surprise_ to first time users.

Reply:
Unless I misunderstand, on Windows "C:/foo" join "/" should give you "C:/" which is still an absolute path.

Reply:
oh ya, that's right! Then I agree that the method, if it should exist at all, should live on PathOps.

I feel pretty sure that I don't want to remove `join`. There are legitimate uses for it, and not having it makes the cost of refactoring to use this library too high.

Reply:
Opened #46